salt/states/zabbix_host.py

```diff
@@ -122,8 +122,8 @@ def _interface_format(interfaces_data):
             interface_type = interface_ports[value['type'].lower()][0]
             main = '1' if six.text_type(value.get('main', 'true')).lower() == 'true' else '0'
             useip = '1' if six.text_type(value.get('useip', 'true')).lower() == 'true' else '0'
-            interface_ip = value.get('ip')
-            dns = value.get('dns', key)
+            interface_ip = value.get('ip') or ""
+            dns = value.get('dns', key) or ""
```

Comment:
Why is the `or` statement necessary here?

Reply:
My SLS specified "- dns" but nothing for "- ip", which generated this for Zabbix API params:
```
		"interfaces": [{
			"port": "10050",
			"dns": "xxx.yyy",
			"type": "1",
			"useip": "0",
			"ip": null,
			"main": "1"
		}]
```
This caused the Zabbix API to repeatedly fail with invalid parameter to function. After switching to empty strings instead of nulls the API calls began working. I suppose that specifying "- ip: '0.0.0.0'" is a smart idea in any case, but it doesn't seem like that should be a requirement when only using DNS.

Reply:
I see now that you're probably referring to ```dns = value.get('dns', key) or ""```. To be honest, I did not notice the ```key``` argument. You're right, no reason for an or statement there.